dm/tests/shardddl1/run.sh

```diff
@@ -611,6 +611,61 @@ function DM_COMPACT() {
 		"clean_table" ""
 }
 
+function DM_COMPACT_USE_DOWNSTREAM_SCHEMA_CASE() {
+	END=10
+	for i in $(seq 0 $END); do
+		run_sql_source1 "insert into ${shardddl1}.${tb1}(b,c) values($i,$i)"
+		run_sql_source1 "update ${shardddl1}.${tb1} set c=1 where a=$((i * 2 + 100))"
+		run_sql_source1 "update ${shardddl1}.${tb1} set c=c+1 where a=$((i * 2 + 100))"
+		run_sql_source1 "update ${shardddl1}.${tb1} set b=b+1 where a=$((i * 2 + 100))"
+		run_sql_source1 "update ${shardddl1}.${tb1} set a=a+100 where a=$((i * 2 + 100))"
+		run_sql_source1 "delete from ${shardddl1}.${tb1} where a=$((i * 2 + 200))"
+		run_sql_source1 "insert into ${shardddl1}.${tb1}(b,c) values($i,$i)"
+	done
+	run_sql_tidb_with_retry_times "select count(1) from ${shardddl}.${tb};" "count(1): 11" 30
+	run_sql_tidb "create table ${shardddl}.${tb}_temp (a int primary key auto_increment, b int unique not null, c int) auto_increment = 100; 
+	             insert into ${shardddl}.${tb}_temp (a, b, c) select a, b, c from ${shardddl}.${tb};
+				 drop table ${shardddl}.${tb}; rename table ${shardddl}.${tb}_temp to ${shardddl}.${tb};"
+	check_sync_diff $WORK_DIR $cur/conf/diff_config.toml 30
+	downstreamSuccess=$(cat $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log | grep "downstream identifyKey check success" | wc -l)
+	downstreamFail=$(cat $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log | grep "downstream identifyKey check failed" | wc -l)
+	if [[ "$downstreamSuccess" -le 50 || "$downstreamFail" -ne 0 ]]; then
```

Comment:
could you briefly explain where the this 50 (10*5) comes from?

Reply:
There is a error in this test as using `AUTO_INCREMENT`, I will fix this error. And then, this value will a determined value when i use `SkipFlushCompactor=return()` to avoid flush in `compact`.

Reply:
why that's 5 per-loop? you can explain it in test comment

Reply:
when use `DownstreamIdentifyKeyCheckInCompact=return(20)`, it will check whether the key value using in `compact` is greater than `20`. On the surface, as `update` will be split to `delete` and `insert`, the count of sqls gets into `compact` will be `88 = 11*(7+1)`, but some sql will be no use, like `update tb1 set c=1 where a=100` which is behind of  `insert into tb1(a,b,c) values(100,1,1)`. This will be more obvious when use `AUTO_INCREMENT` and it will be also influential for the count of `compact`.

Reply:
as `compact` is affected by `j.tp == flush`, the check count of `compact` maybe still use `-le 50`.